### PR TITLE
chore: Refactor PortalUser type

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-1921
## Proposed changes

* `any` types in mock data are replaced with correct types
* Type errors resulting from the change are resolved
* Functionality in app remains the same

## Reviewer notes

Since this PR is only focused on updating types, there shouldn't be any impact on functionality.

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

**Review by shkeating — APPROVED** [review]

**Review by gidjin — APPROVED** [review]
After merging and resolving the conflicts this PR became small, I assume much of what was here was in #987. @abbyoung I'd like you to check before we merge in but I'm good assuming I did the merge right :)

**Pull request merged**
